etna/auto/auto.py

```diff
@@ -0,0 +1,238 @@
+from typing import Callable
+from typing import List
+from typing import Optional
+from typing import Union
+
+import pandas as pd
+from hydra_slayer import get_from_params
+from optuna.storages import BaseStorage
+from optuna.storages import RDBStorage
+from optuna.trial import Trial
+from typing_extensions import Protocol
+
+from etna.auto.optuna import ConfigSampler
+from etna.auto.optuna import Optuna
+from etna.auto.pool import Pool
+from etna.auto.runner import AbstractRunner
+from etna.auto.runner import LocalRunner
+from etna.datasets import TSDataset
+from etna.metrics import MAE
+from etna.metrics import MSE
+from etna.metrics import SMAPE
+from etna.metrics import MedAE
+from etna.metrics import Metric
+from etna.metrics import Sign
+from etna.metrics.utils import MetricAggregationStatistics
+from etna.metrics.utils import aggregate_metrics_df
+from etna.pipeline import Pipeline
+
+
+class _Callback(Protocol):
+    def __call__(self, metrics_df: pd.DataFrame, forecast_df: pd.DataFrame, fold_info_df: pd.DataFrame) -> None:
+        ...
+
+
+class _Initializer(Protocol):
+    def __call__(self, pipeline: Pipeline) -> None:
+        ...
+
+
+class Auto:
```

Comment:
What about `runs_best` and `stack_result`?

Reply:
Is it replaced with `summary`?

Reply:
`runs_result` -> `summary`
`stack_result` -> `top_k`